beacon-chain/rpc/eth/light-client/handlers.go

```diff
@@ -111,27 +115,67 @@ func (s *Server) GetLightClientUpdatesByRange(w http.ResponseWriter, req *http.R
 		return
 	}
 
-	updates := make([]*structs.LightClientUpdateResponse, 0, len(updatesMap))
-
-	for i := startPeriod; i <= endPeriod; i++ {
-		update, ok := updatesMap[i]
-		if !ok {
-			// Only return the first contiguous range of updates
-			break
+	if httputil.RespondWithSsz(req) {
+		var response []byte
+
+		for i := startPeriod; i <= endPeriod; i++ {
```

Comment:
Please check that the context is still live in any loop or potentially expensive operation during a request. The client could have disconnected early and the work could be terminated early too.

```suggestion
		for i := startPeriod; i <= endPeriod; i++ {
			if ctx.Err() != nil {
				httputil.HandleError(w, "context error: "+ctx.Err(), http.StatusInternalServerError)
			}
```